src/core/components/__tests__/ColorModeToggle.test.tsx

```diff
@@ -0,0 +1,3 @@
+import { describe } from 'vitest';
```

Comment:
havent had time to do any unit-testing goodness myself, so just stubbed all-the-things for now. Hoping to be able to get some implemented myself, but any/everyone is welcome to jump in and help out as they're able in follow-up PRs. :lets-go: